model/Software/Classes/SoftwareArtifact.md

```diff
@@ -26,4 +26,19 @@ such as an electronic file, a software package, a device or an element of data.
 - purpose
   - type: SoftwarePurpose
   - minCount: 0
-
+- /Core/Licensing/concludedLicense
+  - type: /Core/Licensing/LicenseField
+  - minCount: 0
+  - maxCount: 1
+- /Core/Licensing/declaredLicense
```

Comment:
It's still up to an SPDX data creator if they want to explicitly specify NOASSERTION, so I would say it should still be an option.

Reply:
But, could we make that a serialization problem? For the model we could say that `NoAssertion` and `null` are the same. And we could give in serializations the option to use `NoAssertion` in place of `null`

Reply:
maybe not blocking, but I think this builds up dept, which we might not be able to get rid of, without breaking changes (which we should discuss before release)

Reply:
I created https://github.com/spdx/spdx-3-model/issues/289 for that

Reply:
When it comes to concludedLicense or declaredLicense, there are two possibilities (leaving aside `NONE` and an actual license / license combination):
1. explicitly state `NOASSERTION` - this means the field is present and explicitly given as NOASSERTION; or
2. not specified - this means the field is _not_ present

For option 2, this was introduced as an option in SPDX 2.3 -- see e.g. https://spdx.github.io/spdx-spec/v2.3/package-information/#713-concluded-license-field: "If the Concluded License field is not present in a package, it implies an equivalent meaning to NOASSERTION."

This was added because people complained about having to have `LicenseConcluded: NOASSERTION` be listed for every package, file or snippet.

So I would still suggest allowing either serialization outcome. Either it is present as NOASSERTION, or it is absent and a consumer would interpret that silence as being semantically equivalent to NOASSERTION (if they care about it at all).

If that's something that we should add to the `concludedLicense` and `declaredLicense` properties, I can submit a PR to add that.